src/Annotation/Scalar.php

```diff
@@ -4,22 +4,35 @@
 
 namespace Overblog\GraphQLBundle\Annotation;
 
+use \Attribute;
+use Doctrine\Common\Annotations\AnnotationException;
+use Doctrine\Common\Annotations\NamedArgumentConstructorAnnotation;
+
 /**
  * Annotation for GraphQL scalar.
  *
  * @Annotation
  * @Target("CLASS")
  */
-final class Scalar implements Annotation
+#[Attribute(Attribute::TARGET_CLASS)]
+final class Scalar extends Annotation implements NamedArgumentConstructorAnnotation
 {
     /**
      * @var string
      */
-    public string $name;
-
+    public ?string $name;
 
     /**
      * @var string
      */
-    public string $scalarType;
+    public ?string $scalarType;
+    
+    public function __construct(?string $name = null, ?string $scalarType = null, ?string $value = null)
```

Comment:
```suggestion
    public function __construct(string $name = null, string $scalarType = null, string $value = null)
```
Just for the info: if an argument has a default value of null, then you don't have to explicitly declare is as nullable, because it defines itself as nullable implicitly. In other words, these 2 strings are equal.

Now the question whether we should or shouldn't type explicit nullable types is still open for me.